libs/contrib/System/Path.idr

```diff
@@ -352,115 +397,147 @@ export
 
-||| Returns the parent and child.
+||| Split path into components.
+|||
+||| ```idris example
+||| splitPath "/usr/local/etc" == ["/", "usr", "local", "etc"]
+||| splitPath "tmp/Foo.idr" == ["tmp", "Foo.idr"]
+||| ```
+export
+splitPath : String -> List String
+splitPath path = map show $ splitPath' (parse path)
+
+||| Split the path into parent and child.
 |||
 ||| ```idris example
 ||| splitParent "/usr/local/etc" == Just ("/usr/local", "etc")
 ||| ```
 export
 splitParent : String -> Maybe (String, String)
-splitParent p = do (a, b) <- splitParent' (parse p)
-                   pure $ (show a, show b)
+splitParent path =
+  do
+    (parent, child) <- splitParent' (parse path)
+    pure (show parent, show child)
 
-||| Returns the path without its final component, if there is one.
+||| Return the path without its final component, if there is one.
 |||
-||| Returns Nothing if the path terminates in a root or volume.
+||| Returns Nothing if the path terminates by a root or volume.
 export
 parent : String -> Maybe String
-parent p = map show $ parent' (parse p)
+parent path = map show $ parent' (parse path)
 
-||| Returns a list of all the parents of the path, longest first,
-||| self included.
+||| Return the list of all parents of the path, longest first, self included.
 |||
 ||| ```idris example
 ||| parents "/etc/kernel" == ["/etc/kernel", "/etc", "/"]
 ||| ```
 export
 parents : String -> List String
-parents p = map show $ iterate parent' (parse p)
+parents path = map show $ List.iterate parent' (parse path)
 
-||| Determines whether base is either one of the parents of full.
+||| Determine whether the base is one of the parents of target.
 |||
 ||| Trailing separator is ignored.
+|||
+||| ```idris example
+||| "/etc" `isBaseOf` "/etc/kernel"
+||| ```
 export
-startWith : (base : String) -> (full : String) -> Bool
-startWith base full = (parse base) `elem` (iterate parent' (parse full))
-
-||| Returns a path that, when appended onto base, yields full.
+isBaseOf : (base : String) -> (target : String) -> Bool
+isBaseOf base target =
+  let 
+    MkPath baseVol baseRoot baseBody _ = parse base
+    MkPath targetVol targetRoot targetBody _ = parse target
+  in
+       baseVol == targetVol
+    && baseRoot == targetRoot
+    && (baseBody `isPrefixOf` targetBody)
+
+||| Return a path that, when appends to base, yields target.
 |||
-||| If base is not a prefix of full (i.e., startWith returns false),
-||| returns Nothing.
+||| Returns Nothing if the base is not a prefix of the target.
 export
-stripPrefix : (base : String) -> (full : String) -> Maybe String
-stripPrefix base full
-    = do let MkPath vol1 root1 body1 _ = parse base
-         let MkPath vol2 root2 body2 trialSep = parse full
-         if vol1 == vol2 && root1 == root2 then Just () else Nothing
-         body <- stripBody body1 body2
-         pure $ show $ MkPath Nothing False body trialSep
+dropBase : (base : String) -> (target : String) -> Maybe String
```

Comment:
Since this is an export, is the breaking renaming necessary? What is wrong with the previous name?

Reply:
I was trying to introduce `isPrefixOf` for Path, just like List. But later I realized that we already have `isPrefixOf` for `String`, which will cause ambiguity, so I renamed prefix to base. And I personally think `base` is more appropriate for paths.

Reply:
Sounds Good!